olive/passes/olive_pass.py

```diff
@@ -80,7 +81,7 @@ def __init__(
         if self._requires_user_script:
```

Comment:
Yeah, @jambay is right, we need download the resource only when needed to be run

Reply:
Seems, it is not a good idea to make `user_script` and `script_dir` accepting azureml resource. It will bring more code structure changes. Let us only accept local file/folder now.

Reply:
Based current design, the UserModuleLoad is used to check the function name right or not in `user_script`. 
It is not good idea to make it lazy I think. Let us say user may want to know the olive config is right or not before any pass run/evaluation.